src/Provider/AbstractProvider.php

```diff
@@ -357,12 +379,17 @@ protected function fetchProviderData($url, array $headers = [])
             $response = (string) $httpResponse->getBody();
         } catch (HttpAdapterException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", (string) $e->getResponse()->getBody());
-            throw new IDPException(end($raw_response));
+            $response = (string) $e->getResponse()->getBody();
             // @codeCoverageIgnoreEnd
         }
 
-        return $response;
+        $result = $this->parseResponse($response);
```

Comment:
@ramsey this is what I was talking about WRT changing the response type of `fetchProviderData`. Tests updated accordingly, but see my comments on #267 about removing the `getUserFoo` methods due to the limited `UserInterface`.